src/Features/VisualBasic/Portable/ReplacePropertyWithMethods/VisualBasicReplacePropertyWithMethods.vb

```diff
@@ -150,11 +153,43 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.CodeRefactorings.ReplaceMethodWithP
                     fieldReference, generator.IdentifierName(setMethod.Parameters(0).Name)))
             End If
 
+            setMethod = UpdateExplicitInterfaceImplementations([property], setMethod, desiredSetMethodName)
             Dim methodDeclaration = generator.MethodDeclaration(setMethod, desiredSetMethodName, statements)
+
             methodDeclaration = CopyLeadingTriviaOver(propertyStatement, methodDeclaration, ConvertValueToParamRewriter.instance)
             Return methodDeclaration
         End Function
 
+        Private Function UpdateExplicitInterfaceImplementations(
+                [property] As IPropertySymbol,
+                method As IMethodSymbol,
+                desiredName As String) As IMethodSymbol
+
+            Dim updatedExplicitImpl = UpdateExplicitInterfaceImplementation(
+                [property], method.ExplicitInterfaceImplementations.FirstOrDefault(), desiredName)
```

Comment:
(we talked about why it's like this, a limitation of the symbol generation apis, so I signed off on the assumption that you'll add a comment)